src/textual/demo.py

```diff
@@ -189,6 +190,14 @@
 """
 
 
+class Note(Message):
```

Comment:
It's a clever idea to have a message that prints a note. But I think this was to work around that `self.app` returns a base App?

If that's the case. I think we should probably revert to the original, as incentive to come up with a better solution for typing `self.app`

Reply:
I don't understand, sorry!

Reply:
I'm guess you did this because `self.app.add_note` was complaining that there was no `add_note` on App? I think we should probably attempt to solve that issue first, as it crops up everywhere.